akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -173,7 +174,21 @@ object Sink {
    *
    * See also [[Flow.limit]], [[Flow.limitWeighted]], [[Flow.take]], [[Flow.takeWithin]], [[Flow.takeWhile]]
    */
-  def seq[T]: Sink[T, Future[immutable.Seq[T]]] = Sink.fromGraph(new SeqStage[T])
+  def seq[T]: Sink[T, Future[immutable.Seq[T]]] = Sink.fromGraph(new SeqStage[T, Vector[T]])
+
+  /**
+   * A `Sink` that keeps on collecting incoming elements until upstream terminates.
+   * As upstream may be unbounded, `Flow[T].take` or the stricter `Flow[T].limit` (and their variants)
+   * may be used to ensure boundedness.
+   * Materializes into a `Future` of `That[T]` containing all the collected elements.
+   * `That[T]` is limited to the limitations of the CanBuildFrom associated with it. For example, `Seq` is limited to
+   * `Int.MaxValue` elements. See [The Architecture of Scala Collectionss](https://docs.scala-lang.org/overviews/core/architecture-of-scala-collections.html) for more info.
+   * This Sink will cancel the stream after having received that many elements.
+   *
+   * See also [[Flow.limit]], [[Flow.limitWeighted]], [[Flow.take]], [[Flow.takeWithin]], [[Flow.takeWhile]]
+   */
+  def collection[T, That](implicit cbf: CanBuildFrom[Nothing, T, That with immutable.Traversable[_]]): Sink[T, Future[That]] =
```

Comment:
@szeiger quick question for advice -- is this future proof or would break with the new collections in 2.13?

Reply:
To keep things moving forward, I took a look at collection-strawman (the scala collection for 2.13), it seems that the changes for our use case would be minor. A rename from CanBuildFrom to BuildFrom, and then calling it inside the function, instead of using cbf() we should use cbf.newBuilder.

Reply:
Thanks for looking into it. Hmm, I wonder if we can add that overload then or if we have to compile it differently for 2.13... but in any case seems like we should be able to pull it off

Reply:
An alternative more stable between scala collection versions, is to drop the implicit argument cbf, and explicitly ask for `the builder` of the collection to the user. Both 2.12 & 2.13 have a compatible Builder[T, That].

The api `collection` would look like:

`def collection[T, That <: immutable.Traversable[T]](builder: mutable.Builder[T,That]): Sink[T, Future[That]] =
    Sink.fromGraph(new SeqStage[T, That](builder))`

If we use it, it's likely that there would be no break in the transition and no change will be need it. The disadvantage is the extra argument, before it was provided implicitly.

Reply:
I'd vote for keeping the implicit and accept having to deal with it somehow when 2.13 arrives.